src/vivarium_gates_nutrition_optimization/components/intervention.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime, timedelta
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+import scipy.stats
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import SimulantData
+from vivarium.framework.randomness import RESIDUAL_CHOICE
+
+from vivarium_gates_nutrition_optimization.constants import (
+    data_keys,
+    data_values,
+    models,
+)
+
+
+class MaternalInterventions:
+    configuration_defaults = {
+        "intervention": {
+            "scenario": "baseline",
+        }
+    }
+
+    @property
+    def name(self) -> str:
+        return "maternal_interventions"
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        start_date = builder.configuration.time.start
+        self.start_date = datetime(start_date.year, start_date.month, start_date.day)
```

Comment:
I believe there is actually a utility function in `vivarium` (I think, but it could be VPH) that converts a config block to a date-time

Reply:
found it, thanks